Source/Core/VideoCommon/VertexLoaderManager.cpp

```diff
@@ -152,6 +155,25 @@ int RunVertices(int vtx_attr_group, int primitive, int count, DataReader src, bo
 	if (skip_drawing || is_preprocess)
 		return size;
 
+	// Object Removal Code code
+	if (m_LocalCoreStartupParameter.num_object_removal_codes)
+	{
+		if (m_LocalCoreStartupParameter.update)
+		{
+			// Set lock so codes can be enabled/disabled in game without crashes.
+			m_LocalCoreStartupParameter.done = false;
+			for (int current_object_removal_code = 0; current_object_removal_code < m_LocalCoreStartupParameter.num_object_removal_codes; current_object_removal_code++)
```

Comment:
@degasus Yep, you're totally right.  I just checked out the disassembly and the compiler is a lot smarter than I thought it was.  It actually looks even cleaner in the disassembly than those unwieldy for loops!  I'll commit this change once I'm done testing, then figure out the mutex locks.  Thanks so much for your help so far.